### PR TITLE
Stable

### PR DESCRIPTION
**Pull request by anton-karlovskiy**
Rewrite the codebase based on react best practices

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,54 +1,43 @@
...
+import React, { useEffect, lazy, Suspense, Fragment } from 'react';
 import { Route, withRouter } from 'react-router-dom';
 import quicklink from 'quicklink/dist/quicklink.mjs';
 
 import './App.css';
 
-const LazyLanding = lazy(() => import(/* webpackChunkName: "landing" */ './containers/Landing'));
-const LazyItemView = lazy(() => import(/* webpackChunkName: "item-view" */ './containers/ItemView'));
+const LazyLanding = lazy(() => import('./containers/Landing/Landing'));
+const LazySaleProducts = lazy(() => import('./containers/SaleProducts/SaleProducts'));
```
Comment (addyosmani) [inline src/App.js]: I see we dropped the `webpackChunkName` entries from here. Was this intentional?

**Comment by addyosmani** [comment]
Can we check the UX for each route is the same as the current version? e.g

new
image

vs current:

image

new
image

vs old
image

**Comment by addyosmani** [comment]
Fyi `Landing.js` has an import pointing to import './landing.css'; which should probably be `import './Landing.css';` (fixed to get running locally)

File: src/App.js [hunk 2]
```diff
@@ -20,8 +20,8 @@ import quicklink from 'quicklink/dist/quicklink.mjs';
 
 import './App.css';
 
-const LazyLanding = lazy(() => import('./containers/Landing/Landing'));
-const LazySaleProducts = lazy(() => import('./containers/SaleProducts/SaleProducts'));
+const LazyLanding = lazy(() => import(/* webpackChunkName: "landing" */ './containers/Landing/Landing'));
+const LazySaleProducts = lazy(() => import(/* webpackChunkName: "sale-products" */ './containers/SaleProducts/SaleProducts'));
```
Comment (addyosmani) [inline src/App.js]: 👍

**Comment by addyosmani** [comment]
Overall this is looking good so far. Could we sanity check that all async chunks / dynamic chunks are using webpack's magic comments so we can see their names while debugging?

image

**Comment by anton-karlovskiy** [comment]
> Overall this is looking good so far. Could we sanity check that all async chunks / dynamic chunks are using webpack's magic comments so we can see their names while debugging?
> 
> image

I double checked all async/dynamic chunks. They are using webpack's magic comments for sure.
Regarding those number names above, they are 3rd party libraries like react-magnifier etc which I think is not possible to name them.

**Comment by anton-karlovskiy** [comment]
> Fyi `Landing.js` has an import pointing to `import './landing.css';` which should probably be `import './Landing.css';` (fixed to get running locally)

I fixed this by renaming from Landing.css to landing.css.

File: src/App.js (hunk 1 above)
Comment (anton-karlovskiy) [reply src/App.js]: It was a mistake, now fixed it.

**Pull request merged**